cpp/src/rolling/grouped_rolling.cu

```diff
@@ -384,7 +384,7 @@ get_null_bounds_for_timestamp_column(column_view const& timestamp_column,
 
   if (timestamp_column.has_nulls()) {
     auto p_timestamps_device_view = column_device_view::create(timestamp_column);
-    auto num_groups               = group_offsets.size();
+    auto num_groups               = group_offsets.size() + 1;
```

Comment:
This is surprising. Without digging into the details, I presume this is a common pattern of a vector of offsets such that the size of any group should be determined by `group_offsets[i+i] - group_offsets[i]`? In which case I'd actually expect the number of groups to be `group_offsets.size() - 1`.

Reply:
I don't think `num_groups` is an accurate name any more. If it is holding the number of groups then it needs some comments explaining why that is true. If it is not then it needs to be renamed.

Line 399 is also very concerning in relation to this change, and I think is what @jrhemstad is pointing out in his comment.

```
auto group_end             = d_group_offsets[group_label + 1];
```

How are we not walking off the end of `d_group_offsets`? `group_label` is the current index between [0, group_offsets.size() + 1), which means the largest `group_label` value will be `group_offsets.size()` which should already be larger than anything `d_group_offsets` could hold, and now we are going one more to it.  Doesn't this mean that before we were reading data off the end of d_group_offsets by 1 value, and now after this change we are reading it up to 2 values past the end?

Reply:
> In which case I'd actually expect the number of groups to be group_offsets.size() - 1.

@jrhemstad, you're right. It should have been `group_offsets.size()-1`. I'm not sure how I managed to flip it. (I think I muffed that up when redoing it for branch-0.19. Fatigue setting in. :|)

It also looks like I should be re-re-targeting for `branch-0.18`, as per @raydouglass's advice. 

Permission to force-push after retesting?

Reply:
> I don't think num_groups is an accurate name any more.

`num_groups` is what it is. I had this right in the `0.18` version of the fix, and fumbled when redoing it for `branch-0.19`.

Reply:
> mythrocks force-pushed the mythrocks:fix-null-bounds-calculation branch from f4a0bdf to 7ca66d5 9 hours ago

Commit f4a0bdf had this right.